### PR TITLE
Convert architecture api test to pytest

### PR DESCRIPTION
**Pull request by dvagala**
Test results:

``` 
 pytest -v  tests/foreman/api/test_architecture.py                       19:50:50
============================================================= test session starts ================================================
platform linux -- Python 3.7.8, pytest-4.6.3, py-1.9.0, pluggy-0.13.1 -- /home/dvagala/rh/robottelo/virtualEnvRobotello/bin/python
cachedir: .pytest_cache
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/dvagala/rh/robottelo
plugins: mock-1.10.4, services-1.3.1, forked-1.2.0, xdist-1.33.0, cov-2.10.0
collecting ... 2020-08-12 17:50:53 - conftest - DEBUG - Collected 15 test cases
collected 15 items                                                                                                                            

tests/foreman/api/test_architecture.py::test_positive_CRUD PASSED                                            [  6%]
tests/foreman/api/test_architecture.py::test_negative_create_with_invalid_name[0] PASSED                     [ 13%]
tests/foreman/api/test_architecture.py::test_negative_create_with_invalid_name[1] PASSED                     [ 20%]
tests/foreman/api/test_architecture.py::test_negative_create_with_invalid_name[2] PASSED                     [ 26%]
tests/foreman/api/test_architecture.py::test_negative_create_with_invalid_name[3] PASSED                     [ 33%]
tests/foreman/api/test_architecture.py::test_negative_create_with_invalid_name[4] PASSED                     [ 40%]
tests/foreman/api/test_architecture.py::test_negative_create_with_invalid_name[5] PASSED                     [ 46%]
tests/foreman/api/test_architecture.py::test_negative_create_with_invalid_name[6] PASSED                     [ 53%]
tests/foreman/api/test_architecture.py::test_negative_update_with_invalid_name[0] PASSED                     [ 60%]
tests/foreman/api/test_architecture.py::test_negative_update_with_invalid_name[1] PASSED                     [ 66%]
tests/foreman/api/test_architecture.py::test_negative_update_with_invalid_name[2] PASSED                     [ 73%]
tests/foreman/api/test_architecture.py::test_negative_update_with_invalid_name[3] PASSED                     [ 80%]
tests/foreman/api/test_architecture.py::test_negative_update_with_invalid_name[4] PASSED                     [ 86%]
tests/foreman/api/test_architecture.py::test_negative_update_with_invalid_name[5] PASSED                     [ 93%]
tests/foreman/api/test_architecture.py::test_negative_update_with_invalid_name[6] PASSED                     [100%]

=================================================== 15 passed in 28.48 seconds ===================================================
 ```

### CODE REVIEW

File: tests/foreman/api/test_architecture.py [hunk 1]
```diff
@@ -14,119 +14,90 @@
...
-    @tier2
-    def test_positive_add_os(self):
-        """Create an architecture and associate it with an OS.
 
-        :id: 9943063d-34f3-4dbc-a341-474ec781e4d9
+@pytest.fixture(scope="module")
+def module_arch():
+    module_arch = entities.Architecture().create()
+    yield module_arch
+    module_arch.delete()
```
Comment (tstrych) [inline tests/foreman/api/test_architecture.py:39]: Please look at above comment

File: tests/foreman/api/test_architecture.py [hunk 2]
```diff
@@ -14,119 +14,90 @@
...
+    """
 
-        :CaseImportance: Critical
-        """
-        for name in valid_data_list().values():
-            with self.subTest(name):
-                arch = entities.Architecture(name=name).create()
-                self.assertEqual(name, arch.name)
+    # Create
+    name = gen_choice(list(valid_data_list().values()))
```
Comment (tstrych) [inline tests/foreman/api/test_architecture.py:56]: I guess the name should be parametrized the same way as you did in `test_negative_create_with_invalid_name`

File: tests/foreman/api/test_architecture.py [hunk 3]
```diff
@@ -14,119 +14,90 @@
...
-from robottelo.test import APITestCase
 
 
-class ArchitectureTestCase(APITestCase):
-    """Tests for architectures."""
+@pytest.fixture(scope="module")
+def module_os():
+    module_os = entities.OperatingSystem().create()
+    yield module_os
+    module_os.delete()
```
[the remark this answers is not in the document]
Comment (dvagala) [reply tests/foreman/api/test_architecture.py:32]: Thanks for pointing that out

There was little [discussion](https://github.com/SatelliteQE/robottelo/pull/7707#discussion_r401013330) about deleting entities after tests. Now as I'm reading that conversation, It's not clear to me why I decided back then to keep deleting the entities in tests. I guess I just see it as a good practice, but I definitely want to keep consistency in tests. I'm going to send an email to start a broader discussion about this.

Just to point out, I see the same approach from @rplevka [here](https://github.com/SatelliteQE/robottelo/commit/9ad94cb5ff0256361a7884046eacdbbf6c061503) and from @jameerpathan111  [here](https://github.com/SatelliteQE/robottelo/blob/master/tests/foreman/api/test_oscap_tailoringfiles.py).

As far as moving the modules into /api_fixtures.py, yes it makes sense. I'll do it after we settle down the issue above.

File: tests/foreman/api/test_architecture.py (hunk 2 above)
Comment (dvagala) [reply tests/foreman/api/test_architecture.py:56]: IIRC I had a discussion with @rplevka way back, that it's probably not worth to test all the names from the valid_data_list() on every test run. 

I don't know it's possible to use the pytest.parametrize here, becuase that way the whole crud test would be executed for each item in the valid_data_list().

 A lot of these crud end to end tests in robotello do it [this way](https://github.com/SatelliteQE/robottelo/blob/602103047e677d7aeb41c777afdb94a5aa4403df/tests/foreman/api/test_computeresource_azurerm.py#L56). Some [of them](https://github.com/SatelliteQE/robottelo/blob/602103047e677d7aeb41c777afdb94a5aa4403df/tests/foreman/api/test_reporttemplates.py#L104)  loops over the list in the test. Another option would be to keep separate parametrized test just for testing entity creation with names from list.

I would suggest the option with looping over them in the test, but again I don't know if it's something that we want to do every time.

Comment (tstrych) [reply tests/foreman/api/test_architecture.py:56]: Good point :+1:  Thanks for explanation and examples :)

File: tests/foreman/api/test_architecture.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (dvagala) [reply tests/foreman/api/test_architecture.py]: For the reference I'll paste here answer from Jitendra: "In simple words, Tearing down the entities arent interfering with other tests entities would be a waste of time. Only the default settings of a satellite should be teardowned/restored !"

**Comment by dvagala** [comment]
So I've removed entities teardown, and create the module fixture in api_fixtures.py and use it from there. The crud test remains the same, without name parametrization.

**Review by rplevka — APPROVED** [review]
@dvagala @tstrych - I would personally prefer to delete the entities in the fixtures as we never know when they start to interfere.
But I'm happy to merge

**Comment by tstrych** [comment]
@rplevka this was discussed in google chat, we can reopen the discussion if you have different opinion on that :)

**Pull request merged**